gitlab/client.py

```diff
@@ -2,10 +2,12 @@
 
 import os
 import re
+import sys
 import time
 from typing import Any, cast, Dict, List, Optional, Tuple, TYPE_CHECKING, Union
 from urllib import parse
 
+import httpx
```

Comment:
This should probably be in a try/except clause and we should only throw an error if people explicitly want to use httpx. We don't want to pull in a new dependency by default if not used, IMO.

I did something similar for graphql here:

https://github.com/python-gitlab/python-gitlab/blob/b8ca5ffb69dce6a7f2f1eb07aeec2216d6817bc5/gitlab/client.py#L1248-L1258

Reply:
Makes sense. Just on quick question. How can we know upfront (before the package is installed) if a user wants to use one of the packages and not the other?

Reply:
I think we can import at the top initially with a try/except and then later when it needs to be used (e.g. when the user provides the httpx client), if it is `None` we can raise an ImportError.
```python
try:
    import httpx
except ImportError:
    httpx = None

# ...

if not httpx:
     raise ImportError("... please install with python-gitlab[httpx]")
```

We do something similar in our cli modules:
https://github.com/python-gitlab/python-gitlab/blob/9b88132078ed37417c2a45369b4976c9c67f7882/gitlab/v4/cli.py#L421-L430

httpx itself is actually a good example of this kind of pattern with all its optional transports:

https://github.com/encode/httpx/blob/14a1704be507cf435fa84da75c249c2ecd1c3981/httpx/_transports/default.py#L158-L164